tumblr-downloader.go

```diff
@@ -150,67 +154,110 @@ func tumblrGetLikes(httpClient *http.Client, blogIdentifier string, timestamp in
 	return parser
 }
 
-func downloadURL(URL string) {
-	resp, err := http.Get(URL)
-	checkError(err)
+func downloadURL(URL string, SubDir string, AppendExtension string) {
+	fmt.Printf("Downloading %s ... ",URL)
 
-	tokens := strings.Split(URL, "/")
-	fileName := tokens[len(tokens)-1]
+	//tokens := strings.Split(URL, "/")
+	//fileName := tokens[len(tokens)-1]
+	fileName := reURL.FindString(URL)
+	if fileName != "" {
 
-	os.Mkdir(outputFolderName, os.ModePerm)
+		outputFilePath := outputFolderName + string(os.PathSeparator) + SubDir + fileName + AppendExtension
+		os.Mkdir(outputFolderName + string(os.PathSeparator) + SubDir, os.ModePerm)
```

Comment:
It's a good idea to group posts according to posters, but I also would like to provide the option (via an env var) to have a flat file hierarchy or not. Because, with some media viewers it's not possible to recursively go through all the media if you'd like to check them one by one.

However, I can add later that in a separate PR.